### PR TITLE
feat: lazy load remote config

### PR DESCRIPTION
**Pull request by astuyve**
The intention of this PR is to support Serverless removing the entire `/appsec` directory, which contains heavy binaries for DDWAF and RemoteConfig which are unused in Serverless workloads, but contribute several MB of space.

That's not possible today, as this code will panic if appsec is removed. We've tested that [here](https://github.com/DataDog/datadog-lambda-python/pull/321).
 
## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] PR description includes explicit acknowledgement/acceptance of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: ddtrace/internal/writer.py [hunk 1]
```diff
@@ -43,6 +42,12 @@
 from .sma import SimpleMovingAverage
 
 
+should_start_remoteconfig = config._appsec_enabled or asbool(os.environ.get("DD_REMOTE_CONFIGURATION_ENABLED", "true"))
```
Comment (astuyve) [inline ddtrace/internal/writer.py:45]: I don't love this, and am amenable to coalescing these in config if that's preferable.

**Review by majorgreys — DISMISSED** [review]

**Review by gnufede — APPROVED** [review]

File: ddtrace/internal/writer/writer.py [hunk 2]
```diff
@@ -45,6 +44,12 @@
 from .writer_client import WriterClientBase
 
 
+should_start_remoteconfig = config._appsec_enabled or asbool(os.environ.get("DD_REMOTE_CONFIGURATION_ENABLED", "true"))
+
+if should_start_remoteconfig:
+    from ddtrace.appsec._remoteconfiguration import enable_appsec_rc
+
+
```
Comment (mabdinur) [inline ddtrace/internal/writer/writer.py:52]:
```suggestion
```

File: ddtrace/internal/writer/writer.py [hunk 3]
```diff
@@ -716,7 +721,8 @@ def start(self):
 
             # appsec remote config should be enabled/started after the global tracer and configs
             # are initialized
-            enable_appsec_rc()
+            if should_start_remoteconfig:
+                enable_appsec_rc()
```
[the remark this answers is not in the document]
Comment (astuyve) [reply ddtrace/internal/writer/writer.py:725]: Okay, i'll remove the current implementation but I do want to disable this if we're in Lambda globally.

**Review by mabdinur — APPROVED** [review]

**Comment by majorgreys** [comment]
@astuyve Do you want to still get this merged?

**Review by P403n1x87 — APPROVED** [review]

**Comment by astuyve** [comment]
I didn't realize this wasn't merged, I will resolve conflicts and push. Thanks @majorgreys

**Review by emmettbutler — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**